lib/Logger/LogAppenderSyslog.cpp

```diff
@@ -59,16 +60,7 @@ LogAppenderSyslog::LogAppenderSyslog(std::string const& facility,
   if ('0' <= facility[0] && facility[0] <= '9') {
     value = StringUtils::int32(facility);
   } else {
-    CODE* ptr = reinterpret_cast<CODE*>(facilitynames);
-
-    while (ptr->c_name != nullptr) {
-      if (strcmp(ptr->c_name, facility.c_str()) == 0) {
-        value = ptr->c_val;
-        break;
-      }
-
-      ++ptr;
-    }
+    value = find_syslog_facility_by_name(facility.c_str());
```

Comment:
Could you not just have used a C-style cast here:
```
  value = (CODE*) (facilitynames.c_str());
```
? Then we would not need a separate file and an actual C compiler for it?
And then maybe a code comment to explain this?